packages/strapi-plugin-content-type-builder/admin/src/utils/formAPI.js

```diff
@@ -92,6 +96,17 @@ const formsAPI = {
 
     return initShape.shape({ pluginOptions: yup.object().shape(pluginOptionsShape) });
   },
+  sanitizeContentTypeSchema(data, initialData) {
+    let sanitizedData = cloneDeep(data);
+
+    const refData = cloneDeep(initialData);
+
+    this.contentTypeSchemaSanitizers.forEach(cb => {
+      sanitizedData = cb(sanitizedData, refData);
```

Comment:
You're re-affecting sanitizedData on each iteration, is this a wanted behaviour?

Reply:
Yes, we need to let the ability for a plugin to mutate the schema before the form submission.